### PR TITLE
Spring client generator

### PR DESCRIPTION
**Pull request by katmatt**
Initial version which generates a very simple spring RestTemplate based client. 
And we already created an end-to-end test with it 😄

### CODE REVIEW

File: common-codegen/src/main/java/io/vrap/rmf/codegen/common/generator/core/CodeGenerator.java [hunk 1]
```diff
@@ -26,10 +27,24 @@ public final Api getApi() {
         return api;
     }
 
-    public final Flowable<AnyType> getRamlObjects() {
+    /**
+     * Returns the types used in the api spec.
+     *
+     * @return the types
+     */
+    public final Flowable<AnyType> getTypes() {
```
Comment (acbeni) [inline common-codegen/src/main/java/io/vrap/rmf/codegen/common/generator/core/CodeGenerator.java]: Like the new names 🥇

File: common-codegen/src/test/java/io.vrap.rmf.codegen.common/CodeGeneratorTest.java [hunk 2]
```diff
@@ -47,11 +43,11 @@ public void init() {
         }
 
         final ClassLoader classLoader = getClass().getClassLoader();
-        final Path inputPath = null;
+        final Path inputPath = Paths.get("/Users/mkoester/Development/commercetools-importer/api-spec/api.raml");;
```
Comment (acbeni) [inline common-codegen/src/test/java/io.vrap.rmf.codegen.common/CodeGeneratorTest.java]: The tests here contains local paths maybe we should replace them by relative paths, WDYT?

Comment (katmatt) [reply common-codegen/src/test/java/io.vrap.rmf.codegen.common/CodeGeneratorTest.java]: Good catch, I even didn't want to push these changes 😉 But I like your idea and will change it 😄

**Comment by katmatt** [comment]
@acbeni I now changed our test to read the environment variable `GENSRC`. If this env var isn't set it falls back to write the output to a temp folder. Additionally our test is now fetching our example api spec from the classpath and we can use this strategy to put smaller example files in our `src/test/resources` folder. Looking forward to your feedback 😄

**Comment by acbeni** [comment]
Looks great to me 👍

**Pull request merged**